application/dash_application/plot_builder.py

```diff
@@ -84,7 +85,8 @@ def generate(title_text, sorted_data, x_fn, y_fn, axis_text, colourby, hovertext
             )
         )
     traces = []
-    grouped_data = sorted_data.groupby(colourby) #TODO: is this inefficient?
+    grouped_data = sorted_data.groupby([colourby, shapeby]) #TODO: is this
```

Comment:
Could you please explain why you group by shapeby as well?

Reply:
(We spoke offline and determined that `groupby` will soon be made obsolete)